client/rest-high-level/src/test/java/org/elasticsearch/client/RequestTests.java

```diff
@@ -155,4 +165,106 @@ private static void getAndExistsTest(Function<GetRequest, Request> requestConver
         assertNull(request.entity);
         assertEquals(method, request.method);
     }
+
+    public void testIndex() throws IOException {
+        String index = randomAsciiOfLengthBetween(3, 10);
+        String type = randomAsciiOfLengthBetween(3, 10);
+        IndexRequest indexRequest = new IndexRequest(index, type);
+
+        String id = randomBoolean() ? randomAsciiOfLengthBetween(3, 10) : null;
+        indexRequest.id(id);
+
+        Map<String, String> expectedParams = new HashMap<>();
+        long version = randomFrom(Versions.MATCH_DELETED, Versions.NOT_FOUND, randomNonNegativeLong());
+        indexRequest.version(version);
```

Comment:
Would not setting the version at all be valid here? If so, maybe also test this (e.g. rarely?)